### ISSUE TITLE
"/accessories"-API should return actual OCI 1.1 `artifactType` of the artifact

### ISSUE THREAD

**Issue by ChristianCiach** [goharbor/harbor#21344]
**Is your feature request related to a problem? Please describe.**

When listing the accessories of an artifact by using the `/accessories` endpoint of the API, the results currently do not contain the actual `artifactType` of the artifact as specified by the [OCI 1.1 image spec](https://github.com/opencontainers/image-spec/blob/v1.1.0/manifest.md#image-manifest-property-descriptions). The `Accessory` model used by Harbor [has a `type` attribute](https://github.com/goharbor/harbor/blob/9da38ae048a186acb022683b38bdea27dcf030e0/src/pkg/accessory/model/accessory.go#L91), but the value can only be [one of a few predefined constants](https://github.com/goharbor/harbor/blob/9da38ae048a186acb022683b38bdea27dcf030e0/src/pkg/accessory/model/accessory.go#L64), severely limiting its usefulness. 

**Describe the solution you'd like**
Instead of the fallback `subject.accessory` type, I would like to see the actual `artifactType` as returned by the [Referrers-API](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#listing-referrers) as the accessory type. Since this would be a (slightly) breaking change, as an alternative we could add a new `artifact_type` attribute to the `Accessory` model.

Also, it should be possible to query only the artifacts of a specific `artifactType`. Since Harbor v2.12 t[here is a new endpoint](https://github.com/goharbor/harbor/commit/b7f1b594951cbbda06242abbd1ebb507c5cbe2dc) to query **all** artifacts of a project that have a given `artifact_type`, but this only works for the whole project, not when listing the accessories of a given repository. For this, I currently have to query the Referrer-API directly:

```
curl 'https://localhost/v2/{project}/{repo}/referrers/{digest}?artifactType=example/artifact-type'
```

In summary:

* The `/accessories` endpoint should return the actual `artifactType` of the OCI 1.1 artifact.
* The `/accessories` endpoint should support to query by a specific `artifactType`.

**Comment by wy65701436** [comment]
hi @ChristianCiach thanks for reporting this issue. 

Let me clarify, the OCI distribution spec is defined for all the v2 apis, and the /accessories is an endpoint of harbor API which only specifically defined for harbor UI/CLI. 

If you call the referres API which is defined in the distribution 1.1, you will see the following results that contains the artifact_type, like:

```
{
    "schemaVersion": 2,
    "mediaType": "application/vnd.oci.image.index.v1+json",
    "manifests": [
        {
            "mediaType": "application/vnd.oci.image.manifest.v1+json",
            "digest": "sha256:a012de24c611dc8f4e752dd7993cfcda4c7e5844fa3c8e31e297b28dbbddb54a",
            "size": 766,
            "annotations": {
                "created": "2024-12-23T04:49:15Z",
                "created-by": "Harbor",
                "org.opencontainers.artifact.created": "2024-12-23T04:49:15Z",
                "org.opencontainers.artifact.description": "SPDX JSON SBOM"
            },
            "artifactType": "application/vnd.goharbor.harbor.sbom.v1"
        }
```
    ]
}

**Comment by ChristianCiach** [comment]
> Let me clarify: the OCI distribution specification is defined for all v2 APIs, while the /accessories endpoint is specific to the Harbor API and is defined solely for use with the Harbor UI/CLI.

Yes, that's why I am currently using the Referrers-API directly, but I still think the `/accessories` endpoint should include the artifactType of the accessory. I see two use-cases for this:

* https://github.com/goharbor/harbor/issues/21345
* We have a CronJob that regularly scans a Harbor project for images with missing SBOMs and Signatures. The scanned project contains thousands of images, and currently I have to query the Referrers-API **for every single artifact**. I already call the `/artifacts`-Endpoint with `?with_accessory=true` to retrieve the list of artifacts, so returning the artifactType with the Accessories-model would save me tens-of-thousands of API-Requests to the Referrers-API with each run of the cronjob.
